src/app/admin/CandidateResponses.tsx

```diff
@@ -0,0 +1,70 @@
+'use client';
+
+import { useState, useEffect } from 'react';
+import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui';
+
+interface CandidateData {
+    email: string;
+    submission: string | null;
+    submissionTime: number | null;
+    isStarted: boolean;
+    startTime: number | null;
+    endTime: number | null;
+}
+
+export const CandidateResponses = () => {
+    const [candidates, setCandidates] = useState<CandidateData[]>([]);
+
+    useEffect(() => {
+        const fetchCandidates = async () => {
+            try {
+                const response = await fetch('/api/candidate/all');
+                if (response.ok) {
+                    const data = await response.json();
+                    setCandidates(data);
+                } else {
+                    console.error('Failed to fetch candidates');
+                }
+            } catch (error) {
+                console.error('Error fetching candidates:', error);
+            }
+        };
+
+        fetchCandidates();
+        const interval = setInterval(fetchCandidates, 30000); // Refresh every 30 seconds
+
+        return () => clearInterval(interval);
+    }, []);
```

Comment:
Can you replace with using react-query? I'd also replace the refresh interval with a refresh button